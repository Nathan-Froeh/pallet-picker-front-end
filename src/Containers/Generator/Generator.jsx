import React, {Component} from 'react';
import './Generator.scss';
import Color from '../../Components/Color/Color';
import RandomColor from 'randomcolor';

export class Generator extends Component {
  constructor () {
    super ();
    this.state = {
      // randomColors: [
        color_1:{color:'#FFFFFF', isLocked: false},
        color_2:{color:'#FFFFFF', isLocked: false},
        color_3:{color:'#FFFFFF', isLocked: false},
        color_4:{color:'#FFFFFF', isLocked: false},
        color_5:{color:'#FFFFFF', isLocked: false}
      // ]
    }
  }

  getRandomColor = () => {
    let newRandomColors = []
    for(var i = 0; i < 5; i++) {
      newRandomColors.push({color: RandomColor(), isLocked: false})
    }
    this.setState(
      {
        color_1: newRandomColors[0], 
        color_2: newRandomColors[1], 
        color_3: newRandomColors[2], 
        color_4: newRandomColors[3], 
        color_5: newRandomColors[4] 
      })
    return this.state.randomColors;
  }

  clearRandomColor = () => {
    this.setState({randomColors: []})
    this.getRandomColor()
  }

  // toggleLock = (id) => {
  //   this.setState({randomColors})
  // }


  render () {
    const {color_1, color_2, color_3, color_4, color_5} = this.state
    return (
      <div className='generator'>
        <h2>Palette Generator Goes here</h2>
        <button onClick={this.clearRandomColor}>
          Generate New Palette
        </button>
        <section className='color--container'>
          <Color hexCode={color_1.color} key={1}/>
          <Color hexCode={color_2.color} key={2}/>
          <Color hexCode={color_3.color} key={3}/>
          <Color hexCode={color_4.color} key={4}/>
          <Color hexCode={color_5.color} key={5}/>
        </section>
        <form action="">
          <select name="Select project">
            <option value="Test 1">Test 1</option>
            <option value="Test 2">Test 2</option>
            <option value="Test 2">Test 3</option>
          </select>
          <input type="text"/>
          <input type="submit"/>
        </form>
      </div>
    )
  }
}

export default Generator;