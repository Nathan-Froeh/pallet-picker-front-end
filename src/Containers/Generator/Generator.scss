.color-container {
  min-width: 100%;
  display: flex;
  overflow-x: auto;
}

.lock {
  margin: 5px auto;
  width: 50%;
}