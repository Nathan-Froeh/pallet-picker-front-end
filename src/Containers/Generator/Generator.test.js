import React from 'react';
import {shallow} from 'enzyme';
import Generator from './Generator';

describe('Generator', () => {
  let wrapper, instance;
  const projects = [{name: 'Bat Cave'}]
  const palettes = [{
    name: 'floor',
    color_1: '#000000',
    color_2: '#000000',
    color_3: '#000000',
    color_4: '#000000',
    color_5: '#000000'
  }]
  const props = {
    projects: projects,
    getAllPalettes: jest.fn(() => palettes)
  }
  beforeEach(() => {
    wrapper = shallow(<Generator {...props}/>)
    instance = wrapper.instance()
  })

  it('Generator should match snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })


})