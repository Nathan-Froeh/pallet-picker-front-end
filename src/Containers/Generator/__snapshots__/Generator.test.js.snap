// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Generator Generator should match snapshot 1`] = `
<div
  className="generator"
>
  <section
    className="color-container"
  >
    <Color
      hexCode="#000000"
      id={0}
      isLocked={false}
      key="0"
      toggleLock={[Function]}
    />
    <Color
      hexCode="#000000"
      id={1}
      isLocked={false}
      key="1"
      toggleLock={[Function]}
    />
    <Color
      hexCode="#000000"
      id={2}
      isLocked={false}
      key="2"
      toggleLock={[Function]}
    />
    <Color
      hexCode="#000000"
      id={3}
      isLocked={false}
      key="3"
      toggleLock={[Function]}
    />
    <Color
      hexCode="#000000"
      id={4}
      isLocked={false}
      key="4"
      toggleLock={[Function]}
    />
  </section>
  <button
    className="gen-palette"
    onClick={[Function]}
  >
    Generate New Palette
  </button>
  <form
    id="save-palette"
    onSubmit={[Function]}
  >
    <select
      name="Select project"
      onChange={[Function]}
      required={true}
    >
      <option
        value=""
      >
        --Choose a project--
      </option>
      <option
        key="0"
        value="Bat Cave"
      >
        Bat Cave
      </option>
    </select>
    <input
      placeholder="New palette name"
      type="text"
    />
    <input
      type="submit"
      value="Save palette"
    />
  </form>
</div>
`;
